### PR TITLE
retroshare: init at 6.6.0

### PR DESCRIPTION
**Pull request by Stekke**
###### Motivation for this change
This packaged got dropped in #123345, however I was in the process of updating it to the recently released 6.6.0
cc @dotlambda, @Reilithion

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124177` at e8bcaa22 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>retroshare</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `opencv`.
Near pkgs/applications/networking/p2p/retroshare/default.nix:2:54:
```
  |
2 | , speex, libmicrohttpd, libxml2, libxslt, sqlcipher, opencv, rapidjson
  |                                                      ^
```
</li>
  </ul>
</details>

File: pkgs/applications/networking/p2p/retroshare/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  ];
+
+  qmakeFlags = [
+    # Upnp library autodetection doesn't work
+    "RS_UPNP_LIB=miniupnpc"
+
+    # These values are normally found from the .git folder
+    "RS_MAJOR_VERSION=6"
+    "RS_MINOR_VERSION=6"
+    "RS_MINI_VERSION=0"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/p2p/retroshare/default.nix:37]: Please get them from the version via lib.versions.

Comment (Stekke) [reply pkgs/applications/networking/p2p/retroshare/default.nix:37]: Thanks! Didn't knew about that.

File: pkgs/applications/networking/p2p/retroshare/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  ];
+
+  postInstall = ''
+    # BT DHT bootstrap
+    cp libbitdht/src/bitdht/bdboot.txt $out/share/retroshare
+  '';
+
+  meta = with lib; {
+    description = "Decentralized peer to peer chat application.";
+    homepage = "http://retroshare.sourceforge.net/";
```
Comment (dotlambda) [inline pkgs/applications/networking/p2p/retroshare/default.nix:48]: Shouldn't this be https://retroshare.cc/index.html?

Comment (Stekke) [reply pkgs/applications/networking/p2p/retroshare/default.nix:48]: Good catch (they used to be on sourceforge)

**Pull request merged**